client/parser/format.go

```diff
@@ -0,0 +1,737 @@
+package parser
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type varDefState int
```

Comment:
@daveyarwood  This is the main piece of complexity that I think makes the formatter a bit spaghetti. I couldn't think of a better way to do it though. The fact is, it properly encapsulates the logic of what we want. Only during the last node of a var def should we allow indentation to wrap and continue onto the next line. It's a bit tricky, but I think it's safe enough for our purposes.